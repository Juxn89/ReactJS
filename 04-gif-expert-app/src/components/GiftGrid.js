import React, { useEffect, useState } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from 'prop-types';

const GiftGrid = ({category}) => {
    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__bounce animate__flash">Loading</p>}
            <div className="card-grid">
                { 
                    images.map( img => (
                        <GiftGridItem 
                            key={img.id} 
                            {...img}
                        />
                    )) 
                }
            </div>
        </>
    );
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GiftGrid;