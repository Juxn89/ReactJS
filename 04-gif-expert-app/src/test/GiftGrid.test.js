import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import React from 'react';
import GiftGrid from '../components/GiftGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs'); // simula una llamada al archivo indicado en la ruta.

describe('Test on GiftGrid', () => {
    const category = 'Makoto Shishio';

    test('Should show it correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GiftGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show items when is loaded useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhos/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GiftGrid category={category} />);
        //expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GiftGridItem').length).toBe(gifs.length);
    });
});