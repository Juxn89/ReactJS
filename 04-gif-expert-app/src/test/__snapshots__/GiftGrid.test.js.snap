// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftGrid Should show it correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__fadeIn"
  >
    Makoto Shishio
  </h3>
  <p
    className="animate__animated animate__bounce animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
